src/flb_utils.c

```diff
@@ -585,9 +585,12 @@ int flb_utils_write_str(char *buf, int *off, size_t size,
         }
         else if (c >= 0x80 && c <= 0xFFFF) {
             hex_bytes = flb_utf8_len(str + i);
-            if ((available - written) < 6) {
+            if (available - written < 6) {
                 return FLB_FALSE;
             }
+            if (i + hex_bytes > str_len) {
+                break; /* skip truncated UTF-8 */
```

Comment:
Perhaps `return FLB_FALSE;`?

We failed to convert all the input?

Reply:
My understanding is that returning `FLB_FALSE` here implies "output buffer
is too small", and will make clients retry again with a larger buffer.

Since "truncated UTF-8 bytes" is not something retryable, I think we should not
return `FLB_FALSE` here (or it can result in an infinite loop).

Reply:
Ah, I see.  Thanks!